test/testsuites/defaultcatsrctests.go

```diff
@@ -143,69 +151,68 @@ func testDefaultCatsrcWhileDisabled(t *testing.T) {
+	err = helpers.UpdateCatalogSourceWithRetries(client, &customCatsrc)
+	require.NoError(t, err, "failed to update the CatalogSource custom resource")
 
 	err = wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
-		updatedCatsrc, nil := checkForCatsrc("redhat-operators", namespace)
+		updatedCatsrc, err := checkForCatsrc(catSrcName, namespace)
 		if err != nil || !defaults.AreCatsrcSpecsEqual(&customCatsrc.Spec, &updatedCatsrc.Spec) {
 			return false, err
 		}
 		return true, nil
 	})
 	require.NoError(t, err, "The update on the custom CatalogSource was reverted back")
 
-	customCatsrc, err = checkForCatsrc("redhat-operators", namespace)
-	require.NoError(t, err, "Custom CatalogSource redhat-operators was removed from the cluster after marketplace was restarted")
+	customCatsrc, err = checkForCatsrc(catSrcName, namespace)
+	require.NoError(t, err, "Custom CatalogSource %s was removed from the cluster after marketplace was restarted", catSrcName)
 
-	err = toggle(t, 4, false, false) //Re-enable all default CatalogSources
+	err = toggle(t, disableAllIndex, false, false) //Re-enable all default CatalogSources
 	require.NoError(t, err, "Could not enable default CatalogSources")
 
-	err = checkCreated(4, namespace)
+	err = checkCreated(disableAllIndex, namespace)
 	require.NoError(t, err, "Default CatalogSources were not created properly")
 
 	err = helpers.InitCatSrcDefinition()
 	require.NoError(t, err, "Could not get a default CatalogSource definitions from disk")
 
 	for _, catsrcDef := range helpers.DefaultSources {
-		if catsrcDef.Name == "redhat-operators" {
-			err := wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
-				clusterCatsrc := &olm.CatalogSource{}
-				err = client.Get(context.TODO(), types.NamespacedName{Name: "redhat-operators", Namespace: namespace}, clusterCatsrc)
-				if err != nil || !defaults.AreCatsrcSpecsEqual(&clusterCatsrc.Spec, &catsrcDef.Spec) {
-					return false, err
-				}
-				return true, nil
-			})
-			require.NoError(t, err, "Default CatalogSource was not restored properly")
-			break
+		if catsrcDef.Name != catSrcName {
+			continue
 		}
+		err := wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
+			clusterCatsrc := &olm.CatalogSource{}
+			err = client.Get(context.TODO(), types.NamespacedName{Name: catSrcName, Namespace: namespace}, clusterCatsrc)
+			if err != nil || !defaults.AreCatsrcSpecsEqual(&clusterCatsrc.Spec, &catsrcDef.Spec) {
+				return false, err
+			}
+			return true, nil
+		})
+		require.NoError(t, err, "Default CatalogSource was not restored properly")
 	}
 }
 
 // checkForCatsrc checks if CatalogSource is present, and is not being removed after some time has passed
 func checkForCatsrc(name, namespace string) (olm.CatalogSource, error) {
 	client := test.Global.Client
-	// Wait for a minute
-	wait.Poll(time.Second*5, time.Minute*1, func() (done bool, err error) {
-		return false, nil
-	})
+	time.Sleep(60 * time.Second)
```

Comment:
Another minor detail, but there's likely a better implementation than just blindly waiting a minute for changes to propagate.